### ISSUE TITLE
Pressing “Stage All” doesn’t stage the submodule commit update

### ISSUE THREAD

**Issue by richardtop** [git-up/GitUp#809]
![Screen Shot 2022-03-18 at 11 43 25](https://user-images.githubusercontent.com/8013017/158979842-7c6c3fd2-f7b5-476f-a615-38c7826e8f39.jpg)

**Comment by lucasderraugh** [comment]
@richardtop Did you make uncommitted changes to `openssl`? I find that if I'm just changing the submodule commit, then stage all works fine. However, if I have uncommitted changes in the submodule, and try to stage it from the parent, it won't work because we can't commit on behalf of the submodule.

I do find working with submodules a bit clunky in GitUp, but if you have practical suggestions around what I just stated above I'm all ears.

**Comment by richardtop** [comment]
Yeah, I think I've made some changes to the submodule. I think, in this case, "discard all" should either discard all of the changes to the submodule, or at least show some warning.

Another solution in case of a submodule having uncommitted changes is to show a button "open in GitUp" to open another instance of GitUp with that repo, so that it's possible to discard changes there.

**Comment by lucasderraugh** [comment]
A recursive discard could work, but I don't know about a stage as the changes would need to be committed.

It's a bit of a balance as I don't want to always interrupt the flows when people make changes in the current repo and want to commit those but don't need the submodule changes committed just yet.

On a side note, there is Repository > Open Submodule… > [submodule] where you can open that in a new window (or option-key will open in current window). You can also do it from the gear in the top right when you select the submodule in the commit view.

**Comment by richardtop** [comment]
> On a side note, there is Repository > Open Submodule… > [submodule] where you can open that in a new window (or option-key will open in current window). You can also do it from the gear in the top right when you select the submodule in the commit view.

I think it's better to have a button next to the submodule or in the button row "Discard all" "Stage all", or somewhere nearby that allows opening a submodule git repo in one click
